### PR TITLE
dns: fix defunct fd bug in apple resolver

### PR DESCRIPTION
**Pull request by junr03**
Commit Message: fix defunct fd bug in apple resolver.
Risk Level: med - necessary fix for apple-based envoy deploys.
Testing: local testing in Envoy Mobile with lifecycle events that force fd failures. This could be better tested if the Apple API was wrapped and replaced in tests to ensure that the AppleDnsResolverImpl code has defensive logic against given API behavior. This could be done in a subsequent PR.

Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

**Comment by junr03** [comment]
> This could be done in a subsequent PR.

My and @rebello95's thinking is that we can get this PR merged, and the bug fixed for users of this resolver. And I can work on the (kind of unorthodox) additional unit tests in parallel; I can probably get a rough draft of the PR tomorrow.

File: source/common/network/apple_dns_impl.cc [hunk 1]
```diff
@@ -71,9 +71,16 @@ void AppleDnsResolverImpl::initializeMainSdRef() {
 }
 
 void AppleDnsResolverImpl::onEventCallback(uint32_t events) {
-  ENVOY_LOG(debug, "DNS resolver file event");
+  ENVOY_LOG(debug, "DNS resolver file event ({})", events);
   ASSERT(events & Event::FileReadyType::Read);
-  DNSServiceProcessResult(main_sd_ref_);
+  DNSServiceErrorType error = DNSServiceProcessResult(main_sd_ref_);
```
Comment (junr03) [inline source/common/network/apple_dns_impl.cc:76]: I verified that _all_ other calls to the DNSService API that return `DNSServiceErrorType` are handled.

**Review by mattklein123 — DISMISSED** [review]
LGTM. Better tests would be appreciated.

**Comment by mattklein123** [comment]
CI failure looks legit.

/wait

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**